index.js

```diff
@@ -121,26 +124,45 @@ module.exports = {
   }
 };
 
-// Path to the template that contains the shim wrapper around the browser
-// polyfill
+// Path to the template that contains the shim wrapper around the browser polyfill
 const templatePath = path.resolve(__dirname + '/assets/browser-fetch.js.t');
 
 
 // Returns a tree containing the browser polyfill (from `whatwg-fetch` and `abortcontroller-polyfill`),
 // wrapped in a shim that stops it from exporting a global and instead turns it into a module
 // that can be used by the Ember app.
 function treeForBrowserFetch() {
+  const abortcontrollerNode = new Rollup(path.dirname(path.dirname(require.resolve('abortcontroller-polyfill'))), {
+    rollup: {
+      input: 'src/abortcontroller-polyfill.js',
+      output: {
+        file: 'abortcontroller.js',
+        // abortcontroller is polyfill only, the name is only required by rollup iife
+        name: 'AbortController',
+        format: 'iife'
+      }
+    }
+  });
   // Fork whatwg-fetch to provide umd build before official release, no extra change made.
   // We will get back to the official one when new version released.
-  const fetchTree = path.dirname(require.resolve('@xg-wang/whatwg-fetch'));
-  const abortcontrollerTree = path.dirname(require.resolve('abortcontroller-polyfill'));
-  const polyfillTree = concat(new MergeTrees([abortcontrollerTree, fetchTree]), {
-    inputFiles: ['abortcontroller-polyfill-only.js', 'fetch.umd.js'],
+  const fetchNode = new Rollup(path.dirname(path.dirname(require.resolve('@xg-wang/whatwg-fetch'))), {
+    rollup: {
+      input: 'fetch.js',
+      output: {
+        file: 'fetch.js',
+        name: 'WHATWGFetch',
+        format: 'iife'
```

Comment:
Shouldn't we be supplying `amd` instead of `iife` for `treeForVendor()`?

Reply:
Because the exported module for app to do import is `fetch` (ember-fetch) rather than this `fetch` (WHATWGFetch). 
We are exporting `ember-fetch`'s amd module, inside the module loading process of `ember-fetch` we need to load the polyfills, so they should be iife but not a module that exported as amd for app to import.